pkg/importer/dataStream.go

```diff
@@ -137,17 +169,25 @@ func CopyImage(dest, endpoint, accessKey, secKey string) error {
 	if err != nil {
 		return errors.Wrapf(err, "unable to create data stream")
 	}
-	defer CloseReaders(ds.Readers)
+	defer ds.Close()
 	return ds.copy(dest)
 }
 
-// Parse the endpoint extensions and return a slice of read-closers. The readers are in order starting with
-// the lowest level reader (eg. http) with the last entry being the highest level reader (eg, gzip).
-// Note: the .qcow2 ext is ignored however qemu files are detected based on their magic number.
-// Note: readers are not closed here.
-func (d *dataStream) constructReaders() ([]io.ReadCloser, error) {
+// Parse the endpoint extensions and return a slice of read-closers. The readers order starts with the
+// lowest level reader (eg. http). The next reader is the final reader in the chain (eg. gz). The last
```

Comment:
> The next reader is the final reader in the chain (eg. gz).

This is true when there is 1 level of compression. For more than 1, this is not true. It might be more descriptive to say that the readers are indexed in the unpacking order of operation.